### PR TITLE
Use pageRange in PageBlob.

### PR DESCRIPTION
**Pull request by sima-zhu**
PageBlobClient mixused of pageRange and blobRange.

Change the blobRange in PageBlob to pagePange.

Reference: https://docs.microsoft.com/en-us/rest/api/storageservices/operations-on-page-blobs

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java [hunk 1]
```diff
@@ -373,24 +372,28 @@
...
-        return withContext(context -> getPageRangesWithResponse(blobRange, accessConditions, context));
+        return withContext(context -> getPageRangesWithResponse(pageRange, accessConditions, context));
     }
 
-    Mono<Response<PageList>> getPageRangesWithResponse(BlobRange blobRange, BlobAccessConditions accessConditions,
+    Mono<Response<PageList>> getPageRangesWithResponse(PageRange pageRange, BlobAccessConditions accessConditions,
         Context context) {
-        blobRange = blobRange == null ? new BlobRange(0) : blobRange;
+        if (pageRange == null) {
+            // Throwing is preferred to Single.error because this will error out immediately instead of waiting until
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: Why are we throwing an error here now? Also, if we keep this let's not use RxJava terminology.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java [hunk 2]
```diff
@@ -421,25 +424,30 @@
...
 
-    Mono<Response<PageList>> getPageRangesDiffWithResponse(BlobRange blobRange, String prevSnapshot,
+    Mono<Response<PageList>> getPageRangesDiffWithResponse(PageRange pageRange, String prevSnapshot,
         BlobAccessConditions accessConditions, Context context) {
-        blobRange = blobRange == null ? new BlobRange(0) : blobRange;
+        if (pageRange == null) {
+            // Throwing is preferred to Single.error because this will error out immediately instead of waiting until
+            // subscription.
+            throw logger.logExceptionAsError(new IllegalArgumentException("pageRange cannot be null."));
+        }
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: Same question here, if it is because there is no constructor for ’PageRange’ like ’BlobRange’ is there a possible substitute?

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/AppendBlobAPITest.groovy [hunk 3]
```diff
@@ -161,6 +161,7 @@ class AppendBlobAPITest extends APISpec {
 
     def "Append block defaults"() {
         setup:
+
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/AppendBlobAPITest.groovy]: Is this change needed?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java (hunk 2 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: PageRange required to specify the start and end. I don't think we need to set default to it.
https://docs.microsoft.com/en-us/rest/api/storageservices/put-page

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/AppendBlobAPITest.groovy (hunk 3 above)
Comment (sima-zhu) [reply sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/AppendBlobAPITest.groovy]: No, will revert. Will update tests in another commit.

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: PutRange needs this check, but get pages doesn't (which this uses).

https://docs.microsoft.com/en-us/rest/api/storageservices/get-page-ranges

Comment (sima-zhu) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/PageBlobAsyncClient.java]: Make sense.

**Comment by sima-zhu** [comment]
Close the PR as the changes do not apply according to the rest API doc.
